packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -122,19 +122,30 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
   await progressHandler?.next(DeployArmTemplatesSteps.DeployArmTemplates);
 
   const azureInfraDir = path.join(ctx.root, baseFolder);
+  generateResourceName(ctx);
 
   // update parameters
-  const parameterTemplate = await fs.readFile(
-    path.join(azureInfraDir, parameterFolder, parameterTemplateFileName),
-    ConstantString.UTF8Encoding
-  );
-  const parameterJson = JSON.parse(expandParameterPlaceholders(ctx, parameterTemplate));
   const parameterDefaultFilePath = path.join(
     azureInfraDir,
     parameterFolder,
     parameterDefaultFileName
   );
-  await fs.writeFile(parameterDefaultFilePath, parameterJson);
+  let parameterTemplate, parameterJsonString;
+  try {
+    await fs.stat(parameterDefaultFilePath);
```

Comment:
What about check whether the `parameterDefaultFilePath` exists first. If yes, read the file. Else, read the parameterTemplateFile.

Reply:
What if fail to read parameter from these two files?